cmd/juju/cloud/updatecloud.go

```diff
@@ -145,20 +138,11 @@ func (c *updateCloudCommand) Run(ctxt *cmd.Context) error {
 		}
 		c.Cloud = r.cloudName
 	}
-
-	if c.BothClientAndController || c.ControllerOnly {
-		if c.ControllerName == "" {
-			// The user may have specified the controller via a --controller option.
-			// If not, let's see if there is a current controller that can be detected.
-			var err error
-			c.ControllerName, err = c.MaybePromptCurrentController(ctxt, fmt.Sprintf("update cloud %q on", c.Cloud))
-			if err != nil {
-				return errors.Trace(err)
-			}
-		}
+	if err := c.MaybePrompt(ctxt, fmt.Sprintf("update cloud %q on", c.Cloud)); err != nil {
+		return errors.Trace(err)
 	}
-	if c.ControllerName == "" && !c.ClientOnly {
-		ctxt.Infof("To update cloud %q on this client, use the --client-only option.", c.Cloud)
+	if c.ControllerName == "" && !c.Client {
+		ctxt.Infof("To update cloud %q on this client, use the --client option.", c.Cloud)
```

Comment:
we can return early here etc

Reply:
No, this was information message. And anyway, it's gone now.